src/java.base/share/classes/java/lang/Character.java

```diff
@@ -9478,7 +9477,7 @@ public static int codePointBefore(char[] a, int index) {
      * @since  1.5
      */
     public static int codePointBefore(char[] a, int index, int start) {
-        if (index <= start || start < 0 || start >= a.length) {
+        if (index <= start || start < 0 || start >= a.length || index > a.length) {
```

Comment:
Since `index > start` it is sufficient to throw `index > a.length`.
The check `start >= a.length` is redundant.

`0 <= start < index <= a.length`

Reply:
Done. I think we should rename the ticket as soon as we modify two methods now

Reply:
/sponsor